crypto/pkcs12/p12_mutl.c

```diff
@@ -238,9 +363,119 @@ int PKCS12_set_mac(PKCS12 *p12, const char *pass, int passlen,
     return 1;
 }
 
-/* Set up a mac structure */
-int PKCS12_setup_mac(PKCS12 *p12, int iter, unsigned char *salt, int saltlen,
-                     const EVP_MD *md_type)
+static int pkcs12_pbmac1_key_gen(const char *pass, int passlen,
+                                 unsigned char *salt, int saltlen,
+                                 int id, int iter, int keylen,
+                                 unsigned char *out,
+                                 const EVP_MD *md_type)
+{
+    return PKCS5_PBKDF2_HMAC(pass, passlen, salt, saltlen, iter,
+                           md_type, keylen, out);
+}
+
+static int pkcs12_mdnid2hmacnid(int mdnid)
+{
+    int hmac_nid = NID_undef;
+
+    switch(mdnid) {
+        case NID_sha1:                    hmac_nid = NID_hmacWithSHA1; break;
+        case NID_md5:                     hmac_nid = NID_hmacWithMD5; break;
+        case NID_sha224:                  hmac_nid = NID_hmacWithSHA224; break;
+        case NID_sha256:                  hmac_nid = NID_hmacWithSHA256; break;
+        case NID_sha384:                  hmac_nid = NID_hmacWithSHA384; break;
+        case NID_sha512:                  hmac_nid = NID_hmacWithSHA512; break;
+        case NID_id_GostR3411_94:         hmac_nid = NID_id_HMACGostR3411_94; break;
+        case NID_id_GostR3411_2012_256:   hmac_nid = NID_id_tc26_hmac_gost_3411_2012_256; break;
+        case NID_id_GostR3411_2012_512:   hmac_nid = NID_id_tc26_hmac_gost_3411_2012_512; break;
+        case NID_sha3_224:                hmac_nid = NID_hmac_sha3_224; break;
+        case NID_sha3_256:                hmac_nid = NID_hmac_sha3_256; break;
+        case NID_sha3_384:                hmac_nid = NID_hmac_sha3_384; break;
+        case NID_sha3_512:                hmac_nid = NID_hmac_sha3_512; break;
+        case NID_sha512_224:              hmac_nid = NID_hmacWithSHA512_224; break;
+        case NID_sha512_256:              hmac_nid = NID_hmacWithSHA512_256; break;
+    }
+
+    return hmac_nid;
+}
+
+int PKCS12_set_pbmac1(PKCS12 *p12, const char *pass, int passlen,
```

Comment:
this is not `_set_pbmac1` it's `_set_pbmac1_with_pbkdf2`

Reply:
Renamed. Thanks